wordpress-mobile/WordPress-iOS
# Reader: Use new Reader streams discover endpoint

Fixes #22700
WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/744

## Description

Updates the Discover endpoint to use the new streams API.

## Testing

To test:
- Launch Jetpack and login
- Navigate to the `Discover` feed
- 🔎 **Verify** the first card is a post
- Scroll until you see a blogs recommendation card
- 🔎 **Verify** only three blogs are recommended
- Scroll until you see a tags recommendation card
- 🔎 **Verify** only five tags are recommended

## Regression Notes
1. Potential unintended areas of impact
Should be none. The API is supposed to be a drop-in replacement.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A
PR submission checklist:


Testing checklist:
